File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int[] nums3=new int[nums1.length+nums2.length];
        int i=0, j=0,k=0;
        
        while(i<nums1.length&&j<nums2.length){
            if(nums1[i]>nums2[j])
                nums3[k++]=nums2[j++];
            else
                nums3[k++]=nums1[i++];
        }
        
        while(i<nums1.length)
            nums3[k++]=nums1[i++];
        
        while(j<nums2.length)
            nums3[k++]=nums2[j++];
                
        return (nums3.length%2==0) ? ((double)(nums3[nums3.length/2]+nums3[(nums3.length/2)-1])/2) : nums3[nums3.length/2]; 
    }
}

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        A, B = nums1, nums2
        
        length = len(A) + len(B)
        
        half = length//2 if length%2==0 else (length+1)//2
        
        if len(A)>len(B):
            A,B = B,A
        
        l, r = 0, len(A) - 1
        
        while True:
            mA = (l+r)//2
            mB = half - mA - 2 
            lA = A[mA] if mA >= 0 else float("-infinity")
            lB = B[mB] if mB >= 0 else float("-infinity")
            rA = A[mA+1] if (mA+1)<len(A) else float("infinity")
            rB = B[mB+1] if (mB+1)<len(B) else float("infinity")
            
            if lB <= rA and lA <= rB:
                if length%2 == 0:
                    return (max(lA,lB) + min(rA,rB) )/2
                else:
                    return max(lA,lB)                

            elif lA > rB:
                r = mA - 1
            else:
                l = mA + 1
                
            
            
        